infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -388,7 +398,14 @@ private void failureDetectorScheduler() {
         // Recover if flag is false
         if (!recovered.getAndSet(true)) {
             recover();
+            sequencerBootstrapped = true;
         }
+
+        if (!isSequencerBootstrapped()) {
+            bootstrapPrimarySequencerServer();
```

Comment:
as we discussed these flags need to be attached to the outcome of the previous operation.